contracts/StrategyCurveTwoPool.sol

```diff
@@ -0,0 +1,383 @@
+  ///   sufficiently large to justify the call to harvest()
+  function harvestTrigger(uint256 callCostInWei)
+    public
+    view
+    override
+    returns (bool)
+  {
+    // Should not trigger if strategy is not active (no assets and no debtRatio). This means we don't need to adjust keeper job.
+    if (!isActive()) {
+      return false;
+    }
+
+    StrategyParams memory params = vault.strategies(address(this));
+
+    // harvest no matter what once we reach our maxDelay
+    if (block.timestamp.sub(params.lastReport) > maxReportDelay) {
+      return true;
+    }
+
+    // harvest our credit if it's above our threshold
+    if (vault.creditAvailable() > creditThreshold) {
+      return true;
+    }
+
+    // trigger if we want to manually harvest
+    if (forceHarvestTriggerOnce) {
+      return true;
+    }
+
+    // otherwise, we don't harvest
+    return false;
+  }
+
+  // convert our keeper's eth cost into want, we don't need this anymore since we don't use baseStrategy harvestTrigger
+  function ethToWant(uint256 _ethAmount)
+    public
+    view
+    override
+    returns (uint256)
+  {
+    return _ethAmount;
+  }
+}
+
+contract StrategyCurveTwoPool is StrategyCurveBase {
+  /* ========== STATE VARIABLES ========== */
+  // `targetTokenAddress` is the token we buy with CRV rewards and the token we then
+  // deposit into the curve pool -- USDT or USDC
+  address public targetTokenAddress;
+
+  IERC20 internal usdt;
+  IERC20 internal usdc;
+
+  // Uniswap V3 pool fees
+  uint24 public crvToWethSwapFee = 3000;
+  uint24 public wethToTargetSwapFee = 500;
+
+  uint24 internal constant maxFee = 10000;
+
+  string internal stratName; // set our strategy name here
+
+  /* ============= CONSTRUCTOR =========== */
+  constructor(
+    string memory _stratName,
+    ContractAddress memory _contractsAddress
+  ) public StrategyCurveBase(_contractsAddress) {
+    // Set our strategy's name
+    stratName = _stratName;
+
+    usdt = IERC20(_contractsAddress.usdt);
+    usdc = IERC20(_contractsAddress.usdc);
+
+    // Required strategic-specific approvals
+        usdt.approve(_contractsAddress.pool, type(uint256).max);
+        usdc.approve(_contractsAddress.pool, type(uint256).max);
+
+        // Start off with USDC
+        targetTokenAddress = _contractsAddress.usdc;
+  }
+
+  /* ========== MUTATIVE FUNCTIONS ========== */
+  // these will likely change across different wants.
+
+  function prepareReturn(uint256 _debtOutstanding)
+    internal
+    override
+    returns (
+      uint256 _profit,
+      uint256 _loss,
+      uint256 _debtPayment
+    )
+  {
+        // Claim and get a fresh snapshot of the strategy's CRV balance
+        _claimRewards();
+
+    uint256 crvBalance = crv.balanceOf(address(this));
+
+        // Sell CRV if we have any
+    if (crvBalance > 0) {
+      _sell(crvBalance);
+      uint256 usdtBalance = usdt.balanceOf(address(this));
```

Comment:
I would move lines 291-293 outside of the if statement and use a separate if for usdc/usdt, for instance you could get USDC/USDT donations to strategy, etc

Reply:
sounds good. By now, I've seen or heard the concept of 'donation' a few times. When would this occur? Why would someone or a contract donate their tokens to the strategy?

Reply:
so realistically it's important to at least be accounting for donations _somehow_ so that they don't wreck your strategy in case anyone malicious sent some in. But also, sometimes we might need to sweep something out, sell it, and donate profits to the strategy, and it's nice if the strategy can just recognize the underlying tokens instead of us having to convert all of the way to `want` before sending it in.